rclpy/src/rclpy/exceptions.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__EXCEPTIONS_HPP_
+#define RCLPY__EXCEPTIONS_HPP_
+
+#include <rcl/error_handling.h>
+
+#include <stdexcept>
+#include <string>
+
+namespace rclpy
+{
+
+std::string append_rcl_error(std::string prepend)
+{
+  prepend += ": ";
+  prepend += rcl_get_error_string().str;
+  rcl_reset_error();
+  return prepend;
+}
```

Comment:
Won't including this header in multiple `*.cpp` files that get compiled into the same binary cause a symbol collision?

Reply:
I suppose so. It looks like `inline` would solve the issue ([Explanation 2. 1.](https://en.cppreference.com/w/cpp/language/inline))

Reply:
Even if this gets inlined, I would assume the `RCLError` constructor would still have the issue. Any reason not to create a `.cpp` file for the implementation?

Reply:
No reason other than laziness :upside_down_face:

Reply:
b802e7e33bdd72e2bf5226ae06f3d68dfe14db1b + 49d5ec0a5f7659749d5a2b67db332f0713d3ab8c + d8182ad put made a cpp file for these and added them to `rclpy_common` - which does double duty of also introducing `pybind11` to `rclpy_common`.